packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -235,4 +235,50 @@ export class Span implements api.Span, ReadableSpan {
     }
     return this._ended;
   }
+
+  // Utility function to truncate given value within size
+  // for value type of string, will truncate to given limit
+  // for type of non-string, will return same value
+  private _truncateToLimitUtil(value: string, limit: number): string {
+    if (value.length <= limit) {
+      return value;
+    }
+    return value.substr(0, limit);
+  }
+
+  /**
+   * If the given attribute value is of type string and has more characters than given {@code attributeValueLengthLimit} then
+   * return string with trucated to {@code attributeValueLengthLimit} characters
+   *
+   * If the given attribute value is array of strings then
+   * return new array of strings with each element truncated to {@code attributeValueLengthLimit} characters
+   *
+   * Otherwise return same Attribute {@code value}
+   *
+   * @param value Attribute value
+   * @returns truncated attribute value if required, otherwise same value
+   */
+  private _truncateToSize(value?: SpanAttributeValue): SpanAttributeValue | undefined {
+    const limit = this._spanLimits.attributeValueLengthLimit;
+    // Check limit
+    // undefined limit means do not truncate
+    if (typeof limit !== 'number' || limit <= 0) {
+      // Non-integer and negative values are invalid, so do not truncate
+      api.diag.warn(`Attribute value limit must be positive, got ${limit}`);
+      return value;
+    }
+
+    // String
+    if (typeof value === 'string') {
+      return this._truncateToLimitUtil(value, limit);
+    }
+
+    // Array of strings
+    if (Array.isArray(value)) {
+      return (value as []).map(val => typeof val === 'string' ? this._truncateToLimitUtil(val, limit) : val);
```

Comment:
is the `as []` cast needed here?

Reply:
Yes,
Getting typecaste lint issues otherwise.